src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
put another way, `bindingContext.ValueProvider.ContainsPrefixAsync()` _always_ returns `true` if an exact match exists so the commend and code are incorrect.  could probably make it almost work passing `bindingContext.ModelName + "."` but `CanValueBindAnyModelProperties()` is more careful and works better in some scenarios.